test/framework/workload.go

```diff
@@ -21,3 +21,7 @@ func (w *WorkloadCluster) DeleteCluster(opts ...CommandOpt) {
 	opts = append(opts, withKubeconfig(w.managementClusterKubeconfigFile()))
 	w.deleteCluster(opts...)
 }
+
+func (w *WorkloadCluster) UpgradeClusterWithGitOps(clusterOpts []ClusterE2ETestOpt) {
```

Comment:
Do we need this method?
We should be able to call `UpgradeWithGitOps` directly on a `WorkloadCluster`